website/content/en/docs/getting-started-with-kops/_index.md

```diff
@@ -0,0 +1,195 @@
+shapes. Karpenter makes scheduling and provisioning decisions based on pod
+attributes such as labels and affinity. In other words, Karpenter eliminates
+the need to manage many different InstanceGroups.
+
+kOps manage provisioners through InstanceGroups. Your cluster will already have
+one Provisioner that will contain a suitable set of instance types for Karpenter to
+choose from.
+
+Managing Provisioner resources directly is possible, but not straight-forward. Read
+more about managing provisioners in the [kOPs Karpenter documentation](https://kops.sigs.k8s.io/operations/karpenter/).
+
+## First Use
+
+Karpenter is now active and ready to begin provisioning nodes.
+As mentioned above, you should already have some Karpenter-managed nodes in your cluster used by
+other kOps addons. Create additional pods using a Deployment, and watch Karpenter provision nodes in response.
+
+### Automatic Node Provisioning
+
+This deployment uses the [pause image](https://www.ianlewis.org/en/almighty-pause-container) and starts with zero replicas.
+
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: inflate
+spec:
+  replicas: 0
+  selector:
+    matchLabels:
+      app: inflate
+  template:
+    metadata:
+      labels:
+        app: inflate
+    spec:
+      terminationGracePeriodSeconds: 0
+      containers:
+        - name: inflate
+          image: public.ecr.aws/eks-distro/kubernetes/pause:3.2
+          resources:
+            requests:
+              cpu: 1
+EOF
+kubectl scale deployment inflate --replicas 5
+kubectl logs -f -n kube-system -l karpenter=controller
+```
+
+### Automatic Node Termination
+
+Now, delete the deployment. After 30 seconds,
+Karpenter should terminate the now empty nodes.
+
+```bash
+kubectl delete deployment inflate
+kubectl logs -f -n karpenter -l karpenter=controller 
+```
+
+### Manual Node Termination
+
+If you delete a node with kubectl, Karpenter will gracefully cordon, drain,
+and shut down the corresponding instance. Under the hood, Karpenter adds a
+finalizer to the node object which blocks deletion until all pods are
+drained and the instance is terminated. Keep in mind, this only works for
+nodes provisioned by Karpenter.
+
+```bash
+kubectl delete node $NODE_NAME
+```
+
+This is similar to `kops delete instance $NODE_NAME` except for that kOps will not respect
+`karpenter.sh/do-not-evict`.
+
+## Upgrading a Cluster
+
+kOps is aware of nodes managed by Karpenter and will handle [rolling upgrades](https://kops.sigs.k8s.io/operations/rolling-update/) of those nodes the same way as any other node:
+
+```
+kops upgrade cluster --yes
+kops update cluster --yes
+kops rolling-update cluster --yes
+```
+
+Karpenter-managed InstanceGroups supports setting `maxUnavailable`, but since Karpenter instances do not run in an Auto Scaling Group, setting `maxSurge` will not have any effect.
+
+During rolling updates, `karpenter.sh/do-not-evict` is not respected.
```

Comment:
This makes me very sad. Why can't kops use `k delete node` to roll the instance group?

Reply:
This is entirely a design decision. Now, I landed on consistent behavior within a kOps cluster rather than consistency across Karpenter clusters.
There shouldn't be a problem to run `delete node` instead of `drain node` on Karpenter InstanceGroups. The question is what the user expects and what makes sense. We are in an active user-initiated upgrade process. Typically performed in maintenance windows. We frequently have issues from people who want us to ignore PDBs, or some sort of timeout on PDB violations, so usually it seems like the behavior users want is to get upgrade done as fast and graceful as possible rather than being endlessly blocked by some pod some user put an annotation on (not necessarily knowing the consequence of that).

Anyways, the documentation is status quo and can be changed if users prefer/expect different behavior. Note that we do not respect CAS' safe to evict-flag either.